packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -1939,17 +2031,24 @@ class AnnotatedRegionLayer<T> extends ContainerLayer {
 
   @override
   S find<S>(Offset regionOffset) {
-    final S result = super.find<S>(regionOffset);
-    if (result != null)
-      return result;
-    if (size != null && !(offset & size).contains(regionOffset))
+    final Iterable<S> all = findAll<S>(regionOffset);
+    if (all.isEmpty) {
       return null;
+    }
+    return all.first;
+  }
+
+  @override
+  Iterable<S> findAll<S>(Offset regionOffset) sync* {
```

Comment:
TBH I really dislike sync*, it has very poor performance characteristics. On the other hand, it is a clear win getting to implement findAll in terms of find in a lazy manner

Reply:
Yeah, this was a tradeoff.  It's an implementation detail, though, so we can change it to use lists or something if it turns out to hurt performance.